backend/src/api/users/user.controller.js

```diff
@@ -27,6 +27,13 @@ const updateUsersController = catchAsync(async (req, res, next) => {
   const updatedUser = await UserDB.findByIdAndUpdate(req.user._id, req.body, {
     new: true,
   });
+
+  if (req.user.balance > 0 && req.body.balance) {
```

Comment:
check after update? It would mean nothing here ))

Reply:
fixed